h2o-r/h2o-package/R/models.R

```diff
@@ -3115,20 +3115,51 @@ h2o.deepfeatures <- function(object, data, layer) {
   h2o.getFrame(dest_key)
 }
 
+
+setClass("H2ONode") # Avoid warning about undefined class, H2ONode has circular dependency
```

Comment:
This piece of code is controversal for me. I point it our for you to notice. H2ONode contains slots `left_child` & `right_child`, which contain `H2ONode`. However, terminal nodes do not have these slots filled. That is why I need `H2ONodeOrNULL`. Yet the circular dependency makes it impossible to create the union type before `setClass`.

It works when done after `setClass`, but then R generates a nasty warning. I managed to get rid of any warnings by first defining a class named H2ONode without any slots, then define the union type and then re-set the `H2ONode` class properly with all the slots.